cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -70,7 +70,13 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
     createLocation(namespaceMeta);
 
     if (cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)) {
-      exploreFacade.createNamespace(namespaceMeta.getNamespaceId().toId());
+      try {
+        exploreFacade.createNamespace(namespaceMeta.getNamespaceId().toId());
+      } catch (ExploreException | SQLException e) {
+        // if we failed to create a namespace in explore then delete the earlier created location for the namespace
+        deleteLocation(namespaceMeta.getNamespaceId());
```

Comment:
This is not full proof but better than  nothing I think. The creation happens in order and we do cleanup of earlier so it will handle almost all the cases except few. How do you suggest to do this in finally ? Check if all the creation happened one by one ? We can surely optimize this further but that is not the purpose of this PR so can we keep this now. Ideally, all this should be happening under one transaction.